### ISSUE TITLE
Feature request: add local and reformer

### ISSUE THREAD

**Issue by samvanstroud** [lucidrains/x-transformers#188]
Thanks for this repo. Is there a possibility of adding your existing [local attention](https://github.com/lucidrains/local-attention) and [reformer](https://github.com/lucidrains/reformer-pytorch) implementations here?

I'm hoping they may also be able to be updated to take advantage of the upcoming attention mask support for the meff kernel in PT2.1.

**Comment by lucidrains** [comment]
yeah, I do have plans to make it so one can register custom transformer blocks. probably will be tested with mixture of experts first https://github.com/lucidrains/st-moe-pytorch, but will prob also consider local attention

**Comment by lucidrains** [comment]
I think Reformer had its chance. I don't think lsh attention will ever work that well
